contracts/SnappBase.sol

```diff
@@ -25,12 +26,13 @@ contract SnappBase is Ownable {
     mapping (uint8 => address) public tokenIdToAddressMap;
 
     struct DepositState {
-        bytes32 shaHash;
-        bool applied;
+        uint16 size;                   // Number of deposits in this batch
+        bytes32 shaHash;               // Rolling shaHash of all deposits
+        uint creationBlock;            // Timestamp of batch creation
+        uint appliedAccountStateIndex; // accountState index when batch applied (for rollback)
     }
 
-    uint16 public slotIndex;
-    mapping (uint => DepositState) public depositHashes;
```

Comment:
I would always prefer to work with mappings in such a situation. Especially, since we want to go later to modular representation and erasing old slots.

Reply:
Yes I also like mapping. What should I do now? Try to do the same stuff with a mapping?

Reply:
Yes, I would do the same with the mapping.